framework/SupervisedLearning/ARMA.py

```diff
@@ -268,39 +362,112 @@ def __evaluateLocal__(self,featureVals):
+              zeroedSample = np.zeros((self.notZeroFilterMask.sum(),1))
+              zeroedSample[:,0] = sample
+            correlatedSample = True # placeholder, signifies we've sampled the correlated distribution
+          # reconstruct base signal from samples
+          ## initialize
+          signal = np.zeros(len(self.pivotParameterValues))
+          ## first the data from the non-zero portions of the original signal
+          signal[self.zeroFilterMask] = unzeroedSample[:,corrIndex]
+          ## then the data from the zero portions (if the filter target, don't bother because they're zero anyway)
+          if target != self.zeroFilterTarget:
+            # fix offset since we didn't include zero-filter target in zeroed correlated arma
+            indexOffset = 0 if corrIndex < filterTargetIndex else -1
+            signal[self.notZeroFilterMask] = zeroedSample[:,corrIndex+indexOffset]
+        # if no zero-filtering (but still correlated):
+        else:
+          ## check if sample taken yet
+          if correlatedSample is None:
+            ## if not, do so now
+            correlatedSample = self._generateVARMASignal(self.varmaResult[0],
+                                                         numSamples = len(self.pivotParameterValues),
+                                                         randEngine = self.normEngine.rvs,
+                                                         rvsIndex = 0)
+          # take base signal from sample
+          signal = correlatedSample[:,self.correlations.index(target)]
+      # if NOT correlated
       else:
         result = self.armaResult[target] # ARMAResults object
         # generate baseline ARMA + noise
-        signal = self._generateARMASignal(result,
-                                          numSamples = len(self.pivotParameterValues),
-                                          randEngine = self.normEngine.rvs)
-
+        # are we zero-filtering?
+        if target == self.zeroFilterTarget:
+          sample = self._generateARMASignal(result,
+                                            numSamples = self.zeroFilterMask.sum(),
+                                            randEngine = self.normEngine.rvs)
+          ## if so, then expand result into signal space (functionally, put back in all the zeros)
+          signal = np.zeros(len(self.pivotParameterValues))
+          signal[self.zeroFilterMask] = sample
+        else:
+          ## if not, no extra work to be done here!
+          sample = self._generateARMASignal(result,
+                                            numSamples = len(self.pivotParameterValues),
+                                            randEngine = self.normEngine.rvs)
+          signal = sample
+      # END creating base signal
+      # DEBUGG adding arbitrary variables for debugging, TODO find a more elegant way, leaving these here as markers
+      #returnEvaluation[target+'_0base'] = copy.copy(signal)
```

Comment:
What do you want to do with these DEBUGs?

Reply:
As mentioned in the comment immediately above, these are flagged for a future improvement, and would have to be re-added almost character for character at that time, which would be wasteful.